src/vis_cpu/conversions.py

```diff
@@ -258,7 +267,57 @@ def equatorial_to_eci_coords(ra, dec, obstime, location, unit="rad", frame="icrs
     return pra, pdec
 
 
-def uvbeam_to_lm(uvbeam, freqs, n_pix_lm=63, polarized=False, **kwargs):
+def prepare_beam(
+    uvbeam: UVBeam, polarized: bool = False, use_feed: Literal["x", "y"] = "x"
+) -> UVBeam:
+    """Prepare an imput beam for either interpolation or simulation.
+
+    The point of this function is to take an arbitrary UVBeam (or AnalyticBeam) and
+    do the necessary checks and conversions to convert it to a format that can be
+    interpolated to an (l,m) grid, or passed to vis_cpu. The output beam type is
+    dependent on the input parameters ``polarized`` and ``use_feed``.
+    """
+    use_feed = use_feed.lower()
+    if use_feed not in "xy":
+        raise ValueError("use_feed must be either 'x' or 'y'")
+    use_pol = use_feed * 2
+
+    # Interpolate beam onto cube
+    if polarized:
+        if uvbeam.beam_type != "efield":
+            raise ValueError("Beam type must be efield")
+        uvbeam_ = uvbeam
+    elif uvbeam.beam_type == "efield":
+        uvbeam_ = uvbeam.copy() if isinstance(uvbeam, UVBeam) else deepcopy(uvbeam)
+        # Analytic beams have no concept of feeds, so assume they have a "single" feed
+        if getattr(uvbeam_, "Nfeeds", 1) > 1:
+            uvbeam_.select(feeds=[use_feed])
+
+        uvbeam_.efield_to_power()
```

Comment:
I think we want to call this method with `calc_cross_pols=False` as an argument. I think this should fix the failing test. We're not allowing the use of cross-polarized power beams, so this fix should be OK. That said, this seems a little bit like a corner case that's not caught in the `UVBeam` code&mdash;it looks like the `efield_to_power` method doesn't account for the case where the user asks for power beams when there's only one feed and `calc_cross_pols` isn't set to `False` (in the sense that we should expect the resulting beam to be real-valued, but it doesn't enter a clause that explicitly sets it to be real-valued).

Reply:
If there's only one feed there is no such thing as a cross pol. One feed yields one power beam no matter what.

Reply:
Oh! I just read that more carefully. Yes, I should add some logic to make it real valued if there's only one feed.